src/main/java/org/folio/innreach/domain/service/impl/ContributionCriteriaConfigurationServiceImpl.java

```diff
@@ -0,0 +1,133 @@
+package org.folio.innreach.domain.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.innreach.domain.dto.ContributionCriteriaConfigurationDTO;
+import org.folio.innreach.domain.dto.ContributionCriteriaExcludedLocationDTO;
+import org.folio.innreach.domain.dto.ContributionCriteriaStatisticalCodeBehaviorDTO;
+import org.folio.innreach.domain.entity.ContributionCriteriaConfiguration;
+import org.folio.innreach.domain.entity.ContributionCriteriaExcludedLocation;
+import org.folio.innreach.domain.entity.ContributionCriteriaStatisticalCodeBehavior;
+
+import org.folio.innreach.domain.exception.EntityNotFoundException;
+import org.folio.innreach.domain.service.ContributionCriteriaConfigurationService;
+import org.folio.innreach.mapper.ContributionCriteriaConfigurationMapper;
+import org.folio.innreach.mapper.ContributionCriteriaExcludedLocationMapper;
+import org.folio.innreach.mapper.ContributionCriteriaStatisticalCodeBehaviorMapper;
+import org.folio.innreach.repository.ContributionCriteriaConfigurationRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.persistence.EntityExistsException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+@Log4j2
+@RequiredArgsConstructor
+@Service
+public class ContributionCriteriaConfigurationServiceImpl implements ContributionCriteriaConfigurationService {
+  private static final String TEXT_CONTRIBUTION_CRITERIA_CONFIGURATION_WITH_ID = "Contribution Criteria Configuration with id: ";
+  private final ContributionCriteriaConfigurationRepository contributionCriteriaConfigurationRepository;
+  private final ContributionCriteriaConfigurationMapper contributionCriteriaConfigurationMapper;
+  private final ContributionCriteriaExcludedLocationMapper excludedLocationMapper;
+  private final ContributionCriteriaStatisticalCodeBehaviorMapper statisticalCodeMapper;
+  @Override
+  public ContributionCriteriaConfigurationDTO create(ContributionCriteriaConfigurationDTO criteriaConfigurationDTO) {
+    var centralServerId = criteriaConfigurationDTO.getCentralServerId();
+    contributionCriteriaConfigurationRepository.findById(centralServerId).ifPresent(entity -> {throw new EntityExistsException(TEXT_CONTRIBUTION_CRITERIA_CONFIGURATION_WITH_ID + entity.getCentralServerId() +" already exists.");});
+
+    var entityForSave = contributionCriteriaConfigurationMapper.toEntity(criteriaConfigurationDTO);
+    entityForSave.getStatisticalCodeBehaviors().forEach(statisticalCodeBehavior -> statisticalCodeBehavior.setContributionCriteriaConfiguration(entityForSave));
```

Comment:
Please, use more meaningful variable names, even if it requires more characters.